src/robot/util.cpp

```diff
@@ -180,13 +180,17 @@ InterpolatedPtr planToTSR(
     double timelimit,
     std::size_t maxNumTrials)
 {
+  // Create an IK solver with metaSkeleton dofs.
+  auto ik = InverseKinematics::create(bn);
+  ik->setDofs(metaSkeleton->getDofs());
```

Comment:
I haven't checked whether this would work even for the case that `metaSkeleton` contains bodies from difference skeletons, which I don't think so. Could wait for me to ensure this?

If so, DART should inform this, but let's have a validity check in AIKIDO or at least a comment here.

Reply:
Ah, good point. Let me add a quick check here with a comment. :)

Reply:
Currently `InverseKinematics::setDofs(~)` does assume that you're only passing it DoF objects that belong to the current Skeleton, and it naively grabs whatever their indices are. I think that can be considered a bug (or at least unintuitive behavior) in DART.

I agree with JS that we should change DART to have the `IK::setDofs(~)` skip over any `DegreeOfFreedom*` objects that don't belong to the IK's parent Skeleton.